returnn/tensor/_tensor_extra.py

```diff
@@ -609,6 +610,116 @@ def __repr__(self):
     def __hash__(self):
         return id(self)
 
+    # _TensorMixin.__eq__ is disabled as per the following error in some TF tests:
+    # AssertionError: unhashable type: 'Tensor'.
+    # See CI https://github.com/rwth-i6/returnn/actions/runs/4406240591
+    """
+    def __eq__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        return self.raw_frontend.compare(self, "==", other)
+    """
+
+    def __ne__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        return self.raw_frontend.compare(self, "!=", other)
+
+    def __lt__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        return self.raw_frontend.compare(self, "<", other)
+
+    def __le__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        return self.raw_frontend.compare(self, "<=", other)
+
+    def __gt__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        return self.raw_frontend.compare(self, ">", other)
+
+    def __ge__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        return self.raw_frontend.compare(self, ">=", other)
+
+    def __add__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        if isinstance(other, (int, float, numpy_number)) and other == 0:
+            return self
+        return self.raw_frontend.combine(self, "+", other)
+
+    def __radd__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        if isinstance(other, (int, float, numpy_number)) and other == 0:
+            return self
+        return self.raw_frontend.combine(other, "+", self)
+
+    def __sub__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        if isinstance(other, (int, float, numpy_number)) and other == 0:
+            return self
+        return self.raw_frontend.combine(self, "-", other)
+
+    def __rsub__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        if isinstance(other, (int, float, numpy_number)) and other == 0:
+            return self
+        return self.raw_frontend.combine(other, "-", self)
+
+    def __mul__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        if isinstance(other, (int, float, numpy_number)) and other == 1:
+            return self
+        return self.raw_frontend.combine(self, "*", other)
+
+    def __rmul__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        if isinstance(other, (int, float, numpy_number)) and other == 1:
+            return self
+        return self.raw_frontend.combine(other, "*", self)
+
+    def __truediv__(self: Tensor, other: Union[_frontend_api.RawTensorTypes, Tensor]) -> Tensor:
+        if isinstance(other, (int, float, numpy_number)):
```

Comment:
This check is duplicated here, right? You already do the same check in `combine`. Then just leave it in `combine`, and remove it here?